apps/daimo-mobile/src/logic/onboarding.ts

```diff
@@ -0,0 +1,46 @@
+import { SlotType, getSlotType } from "@daimo/common";
+import { useCallback, useContext, useMemo } from "react";
+
+import { DispatcherContext } from "../action/dispatch";
+import { useNav } from "../common/nav";
+import { Account } from "../model/account";
+
+export function useOnboardingChecklist(account: Account) {
+  const nav = useNav();
+  const dispatcher = useContext(DispatcherContext);
+
+  const hasBackup = account.accountKeys.some(
+    (key) => getSlotType(key.slot) === SlotType.PasskeyBackup
+  );
+
+  const farcasterConnected = !!account.linkedAccounts.find(
+    (a) => a.type === "farcaster"
+  );
+
+  const allComplete = useMemo(() => {
+    return hasBackup && farcasterConnected;
+  }, [hasBackup, farcasterConnected]);
+
+  const handleSecureAccount = useCallback(() => {
+    nav.navigate("Settings");
+    dispatcher.dispatch({ name: "hideBottomSheet" });
+  }, [nav, dispatcher]);
+
+  const handleConnectFarcaster = useCallback(() => {
+    nav.navigate("Settings");
```

Comment:
same here, does the interaction feel better if we just open the farcaster connect bottom sheet when the user loads the settings?

Reply:
I think just opening the sheet might be smoother in this case. The only issue I can think of is that it might be a little weird for the "connect Farcaster" flow to end on a screen that doesn't show the farcaster username (like the Settings screen does)